src/dmd/cparse.d

```diff
@@ -825,7 +825,7 @@ final class CParser(AST) : Parser!AST
             auto tk = peek(&token);
 
             // If ( type-name )
-            if (isTypeName(tk))
+            if (isTypeName(tk) && peek(tk).value == TOK.rightParenthesis)
```

Comment:
I'm not convinced this is going to be enough. What about `a = (b);`?

Reply:
yeah, you're right.

Reply:
I have been toying around with a table of all typenames in current scope, so this check can be replaced with a lookup. But yet to refactor the current `symbols` member field into a CScope struct to replace the save/restore with push/pop.